codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -636,13 +640,22 @@ public void translate(Entry<CharSequence, CharSequence> entry) throws Http2Excep
                 if (COOKIE.equals(name)) {
                     // combine the cookie values into 1 header entry.
                     // https://tools.ietf.org/html/rfc7540#section-8.1.2.5
-                    String existingCookie = output.get(COOKIE);
-                    output.set(COOKIE,
-                               (existingCookie != null) ? (existingCookie + "; " + value) : value);
+                    if (cookies == null) {
+                        cookies = new StringBuilder(value.length());
+                    } else if (cookies.length() > 0) {
+                        cookies.append("; ");
+                    }
+                    cookies.append(value);
                 } else {
                     output.add(name, value);
                 }
             }
         }
+
+        public void translateCookies() {
```

Comment:
I wonder if it would be better to have this code encapsulated in one method and so make it less error-prone to use.

Basically change the `translate(Entry<....>)` method to `translate(Iterable<...>)` and then do the for-loop in there + the code that you added in `translateCookies` at the end. This way it will be impossible to forget adding the cookies at the end.... 

@kevinoliver WDYT ?

Reply:
yeah i went with least code churn but i prefer this.

going to have a new "public" method `translateHeaders(...)` that does all this.